crates/css-syntax-types/src/lib.rs

```diff
@@ -45,48 +43,92 @@ impl From<&Css> for Css {
 }
 
 #[derive(Clone, Debug, Deserialize, Serialize)]
-#[serde(deny_unknown_fields)]
 pub struct AtRule {
-    pub descriptors: BTreeMap<String, AtruleDescriptor>,
-    pub href: Option<Url>,
     pub name: String,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub prose: Option<String>,
-    pub value: Option<String>,
-    pub values: Option<CssValues>,
+    #[serde(rename = "specLink", default, skip_serializing_if = "Option::is_none")]
+    pub spec_link: Option<SpecLink>,
+    #[serde(default)]
+    pub descriptors: BTreeMap<String, AtruleDescriptor>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub syntax: Option<String>,
 }
 impl From<&AtRule> for AtRule {
     fn from(value: &AtRule) -> Self {
         value.clone()
     }
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Function {
+    #[serde(rename = "specLink", default, skip_serializing_if = "Option::is_none")]
+    pub spec_link: Option<SpecLink>,
```

Comment:
The order is different to Type, is that intentional? It would be nice to be somewhat consistent, assuming the order is relevant for serialisation.